src/renderers/dom/fiber/ReactDOMFiberEntry.js

```diff
@@ -368,7 +368,13 @@ var DOMRenderer = ReactFiberReconciler({
       const hostContextDev = ((hostContext: any): HostContextDev);
       validateDOMNesting(null, text, null, hostContextDev.ancestorInfo);
     }
-    var textNode: TextInstance = document.createTextNode(text);
+
+    var doc = rootContainerInstance;
+    if (rootContainerInstance.nodeType !== DOCUMENT_NODE && doc.ownerDocument) {
```

Comment:
Can `rootContainerInstance` even be `DOCUMENT_NODE`?

Reply:
line 101: type Container = Element | Document; 

I think so.